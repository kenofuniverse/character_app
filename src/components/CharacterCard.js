import React from 'react'
import PropTypes from 'prop-types'
import {
  Card
} from 'react-bootstrap'
import { Link } from 'react-router-dom'
import * as moment from 'moment'

const fromNow = (date) => moment(date).fromNow()

const CharacterCard = ({item}) => (
  <Card>
    <Card.Img variant="top" src={item.image} />
    <Card.Body>
      <Card.Title>
        <Link to={`/character/${item.id}`}>{item.name}</Link>
      </Card.Title>
      <Card.Text>
        {item.type}
      </Card.Text>
    </Card.Body>
    <Card.Footer>
      <small className="text-muted">{fromNow(item.created)}</small>
    </Card.Footer>
  </Card>
)

CharacterCard.propTypes = {
  item: PropTypes.object.isRequired,
}

export default CharacterCard
