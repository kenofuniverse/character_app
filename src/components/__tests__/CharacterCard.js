import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router';
import CharacterCard from '../CharacterCard'

describe('CharacterCard component', () => {
  const item = {
    "id": 2,
    "name": "Morty Smith",
    "status": "Alive",
    "species": "Human",
    "type": "",
    "gender": "Male",
    "image": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
    "created": "2017-11-04T18:50:21.651Z"
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <CharacterCard item={item} onSelectItem={() => {}}/>
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('render contents correctly', () => {
    const rendered = shallow(<MemoryRouter><CharacterCard item={item} onSelectItem={() => {}} /></MemoryRouter>)
    expect(rendered.html()).toContain(item.name)
    expect(rendered.html()).toContain(item.image)
  })
})
