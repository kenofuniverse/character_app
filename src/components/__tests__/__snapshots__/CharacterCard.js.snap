// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard component renders correctly 1`] = `
<div
  className="card card"
>
  <img
    className="card-img-top card__image"
    src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      <a
        href="/character/2"
        onClick={[Function]}
      >
        Morty Smith
      </a>
    </div>
    <p
      className="card-text"
    >
      
    </p>
  </div>
  <div
    className="card-footer"
  >
    <div
      className="row"
    >
      <div
        className="col-8"
      >
        <small
          className="text-muted"
        >
          Created 
          a year ago
        </small>
      </div>
      <div
        className="text-right col-4"
      >
        <div
          className="form-check"
        >
          <input
            checked={false}
            className="form-check-input position-static"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
